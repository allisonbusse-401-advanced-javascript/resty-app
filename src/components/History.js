import React from 'react';
// import styles from './Header.css';
import PropTypes from 'prop-types';
import HistoryItem from './HistoryItem';
import styles from './History.css';

const History = ({ historyItems, onHistoryClick }) => {
  const historyElements = historyItems.map((history, index) => (
    <li key={`${index}-${history.url}-${history.method}`} storekey={history.storeKey}>
      <HistoryItem url={history.url} method={history.method}/>
    </li>
  ));

  return (
    <div className={styles.History}>
      <h2>History</h2>
      <ul onClick={onHistoryClick}>
        {historyElements}
      </ul>
    </div>
  );
};

History.propTypes = {
  historyItems: PropTypes.array.isRequired,
  onHistoryClick: PropTypes.func.isRequired
};


export default History;
