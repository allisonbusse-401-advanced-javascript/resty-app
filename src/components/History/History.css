.History {
  width: 500px;
  margin-right: 40px;
  font-family: Arial, Helvetica, sans-serif;

  li {
    list-style: none;
    background-color: #2D5986;
    color: white;
    padding: 2px 6px;
  }
  li:hover {
    background-color: gray;
    color: black;
  }
  ul {
    padding-left: 0;
  }

  li h2 {
    text-transform: uppercase;
  }
}