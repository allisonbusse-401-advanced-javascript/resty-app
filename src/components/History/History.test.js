import React from 'react';
import { shallow } from 'enzyme';
import History from './History';

describe('History component', () => {
  const props = {
    handleHistoryClick: () => {},
    historyItems: []
  };

  it('renders History', () => {   
    const wrapper = shallow(<History {...props}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
