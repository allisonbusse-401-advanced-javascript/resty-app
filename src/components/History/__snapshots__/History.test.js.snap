// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`History component renders History 1`] = `
<div
  className="History"
>
  <h2>
    History
  </h2>
  <ul />
</div>
`;
